CommonConcepts/Plugins/Rhetos.Dom.DefaultConcepts/DataStructure/HardcodedEntityCodeGenerator.cs

```diff
@@ -0,0 +1,39 @@
+﻿/*
+    Copyright (C) 2014 Omega software d.o.o.
+
+    This file is part of Rhetos.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+using Rhetos.Dsl.DefaultConcepts;
+using System.ComponentModel.Composition;
+using Rhetos.Extensibility;
+using Rhetos.Dsl;
+using Rhetos.Compiler;
+
+namespace Rhetos.Dom.DefaultConcepts
+{
+    [Export(typeof(IConceptCodeGenerator))]
+    [ExportMetadata(MefProvider.Implements, typeof(HardcodedEntityInfo))]
+    public class HardcodedEntityCodeGenerator : IConceptCodeGenerator
+    {
+        public void GenerateCode(IConceptInfo conceptInfo, ICodeBuilder codeBuilder)
+        {
+            HardcodedEntityInfo info = (HardcodedEntityInfo)conceptInfo;
+            var codeSnippet = $@"throw new Rhetos.UserException(""It is not allowed to modify hard-coded data in {conceptInfo.GetKeyProperties()}."");";

```

Comment:
Please use UserException constructor with separated message parameters (for easier localization).
For example see CreatedByCodeGenerator.cs.
Code snippet should end with newline and 12 spaces (if I counted correctly) to match the formatting.